### ISSUE TITLE
Identify platform for the program

### ISSUE THREAD

**Issue by nesquena** [MindMakersProject/Curriculum#1]
Identify the program platform:
-  What are the apps and tools used before and during the program?
- What platform is used to manage the application and admissions?
- What platform or tools are used to enable blended online learning?
- What platform is used for communication, announcements and support?
- Review the [Platform overview](https://github.com/MindMakersProject/Curriculum/wiki/Software-Platform)

**Comment by juanmnl** [comment]
@nesquena I made this list for some of the things we could build/setup, i'm not sure if it belongs here.
Let me know.  

I'm also making an UX diagram trying to figure out all the use cases. (i'll let you know when it's done)

```
##MindMakers

###Static Website 
*(public)  
URL: ​*http://www.mindmakersproject.org*​  
Technology:  
- Npm  
- Webpack  
- Bootstrap  
- React  
- [Surge.sh](https://surge.sh) $0 -or- [Firebase](https://firebase.com) $5 https + db  

**Design, development, setup & deploy.**  

###Forum  
*(public/private)  
URL: ​*http://forum.mindmakersproject.org*​  
#####[Discourse](https://www.discourse.org/)  
Technology:  
- Ember.js  
- Rails  
- AWS, Digital Ocean, Heroku or any other rails hosting service  
**Tweaking, config & deploy.**  

###Blog  
*(public)  
URL: ​*http://blog.mindmakersproject.org*​  
#####[Worpress (calypso)](https://developer.wordpress.com/calypso/) 
Technology:  
- React & Wordpress  
**Design, development, setup & deploy.**

#####Medium Publication
URL: *https://www.medium.com/mindmakers*
Technology:  
- Hosted
**Setup only.**

###Courses Platform 
*(private)  
URL: ​*http://student.mindmakersproject.org or http://courses.mindmakersproject.org*​  
- Build our own platform or use some pre-established ones  
**Design, development, config & deploy.**

###Mobile App (exhibit app?)
- Webapp [insert_prefered_framework] + PhoneGap  
- Angular2 + Ionic2  
- Webapp [insert_prefered_framework] + React Native iOS + Android  
**Design, development, config & app/play store deploy.**
```

**Comment by nesquena** [comment]
This is great, thanks for putting this together!

**Comment by isisAnchalee** [comment]
What would you suggest we use a forum for? It seems like Slack is fine for communication, unless I'm missing something :) 

Also for the static site, are you suggesting re-building it with the listed technologies? I'm not opposed to doing that if people are down, but I think we should list out our priorities.

In my opinion, the next biggest decision we need to make is how we're going to host/display the course curriculums. I think we should have something that is open source/easily accessible by people outside of our immediate team.

**Comment by juanmnl** [comment]
There's an UX diagram that goes with it –haven't finished yet, though.  

tl:dr
There are 4 kinds of users - [access]:  
- Admins/devs (registered) - [slack, admin, goldenapp, forum, courses cms, website cms]  
- Teachers/volunteers (registered) - [slack, goldenapp(limited), forum(limited), courses cms(limited)]  
- Students (registered) - [forum, courses cms(limited)]  
- Visitors (non registered) - [public website]  

Slack is good for admins/devs and teachers/volunteers to quickly communicate, but for registered students and teachers i'd suggest a forum like [discourse](https://www.discourse.org/) (Ex: [React's official forum](https://discuss.reactjs.org/top/all). I think it can spin in a digital ocean droplet.  

For the site, i'm already building a landing page+blog-like template for me, that could be easily cloned and customisable; but not replacing the existing until it's finished. With surge, we can have static hosting for $0.  

Agree that the most important thing to develop is the courses section, that's where the UX diagram comes handy as it shows what elements need to be built –login/registration system, course curriculum, student profile view, etc. Which users have access to what.  
We have to think about the app architecture too, as there's video involved, and from previous experiences, video can become quickly an expen$ive tran$action.  

[This could become useful too](https://www.canvaslms.com/k-12/) last time i checked, it was a rails-based education framework.    

For the curriculum alone, i think there is no better place like here :octocat: [github](https://education.github.com/)  

I'll try to finish that UX diagram as soon as possible. :)

**Comment by cherscarlett** [comment]
Are we married to the idea of using WordPress? I find it's not really worth it considering the security flaws if you have a team of engineers who can utilize something that is a little less plug-and-play. I also I am not really crazy about using PHP, with engineers who can utilize ruby, python, node or java instead.

**Comment by jodavaho** [comment]
I know I"m sort of a late-joiner in this discussion, but I think we can't (shouldn't) seek a platform that can do all that. For example, I agree with @juanmnl that github might be a perfect way to distribute lessons, _especially_ for the software classes, as that's a solid piece of technology that anyone in software should know how to use. It also supports wikis, pages, issues, version, and it seems everyone is already using it.

For video lessons, what's wrong with youtube? We could have a private channel, for example?

Question: Are we concerned about students re-distributing materials? If not, then there's not a _huge_ need for locking down access to videos or course materials. My assumption is that workshops and live-mentoring is the best benefit to enrollment. To support that, slack or a simple tool like discourse (or even a stackexchange-like site), would work very well. 

If there's no need for crazy lockdowns or really fancy websites, could we ship the first iteration of courses using just github, a stackexchange clone, and slack?
